Mathlib/Data/Finset/Pointwise.lean

```diff
@@ -261,6 +283,26 @@ theorem inv_insert (a : α) (s : Finset α) : (insert a s)⁻¹ = insert a⁻¹
 #align finset.inv_insert Finset.inv_insert
 #align finset.neg_insert Finset.neg_insert
 
+@[to_additive (attr := simp)]
+lemma sup_inv [SemilatticeSup β] [OrderBot β] (s : Finset α) (f : α → β) :
+    sup s⁻¹ f = sup s fun x ↦ f x⁻¹ :=
+  sup_image ..
+
+@[to_additive (attr := simp)]
+lemma sup'_inv [SemilatticeSup β] {s : Finset α} (hs : (s⁻¹).Nonempty) (f : α → β) :
+    sup' s⁻¹ hs f = sup' s (.of_inv hs) fun x ↦ f x⁻¹ :=
+  sup'_image ..
+
+@[to_additive (attr := simp)]
+lemma inf_inv [SemilatticeInf β] [OrderTop β] (s : Finset α) (f : α → β) :
+    inf s⁻¹ f = inf s fun x ↦ f x⁻¹ :=
+  inf_image ..
+
+@[to_additive (attr := simp)]
+lemma inf'_inv [SemilatticeInf β] {s : Finset α} (hs : (s⁻¹).Nonempty) (f : α → β) :
+    inf' s⁻¹ hs f = inf' s (.of_inv hs) fun x ↦ f x⁻¹ :=
+  inf'_image ..
```

Comment:
For rewriting purposes, can we have the same lemma reversed assuming `s.Nonempty`?